### PR TITLE
Manual merge

### PR DESCRIPTION
**Pull request by greg0ire**
Don't merge, just approve

### CODE REVIEW

File: src/Test/AbstractBlockServiceTestCase.php [hunk 1]
```diff
@@ -11,19 +11,27 @@
 
 namespace Sonata\BlockBundle\Test;
 
+use PHPUnit\Framework\TestCase;
 use Sonata\BlockBundle\Block\BlockContextInterface;
 use Sonata\BlockBundle\Block\BlockContextManager;
 use Sonata\BlockBundle\Block\BlockContextManagerInterface;
 use Sonata\BlockBundle\Block\BlockServiceInterface;
 use Sonata\BlockBundle\Block\BlockServiceManagerInterface;
 use Sonata\BlockBundle\Tests\Block\Service\FakeTemplating;
```
Comment (jordisala1991) [inline src/Test/AbstractBlockServiceTestCase.php]: Remove this use to make test pass

Comment (greg0ire) [reply src/Test/AbstractBlockServiceTestCase.php]: Thanks for the hint buddy!

Comment (greg0ire) [reply src/Test/AbstractBlockServiceTestCase.php]: Hmm but that breaks :P

Comment (greg0ire) [reply src/Test/AbstractBlockServiceTestCase.php]: Oh sorry I thought you meant the `TestCase one

**Comment by greg0ire** [comment]
It's better but I still get one failure:


```
Time: 275 ms, Memory: 20.00MB

There was 1 failure:

1) Sonata\BlockBundle\Tests\Block\Service\MenuBlockServiceTest::testBuildEditForm
Sonata\BlockBundle\Menu\MenuRegistryInterface::getAliasNames() was not expected to be called more than once.

/home/greg/dev/SonataBlockBundle/src/Block/Service/MenuBlockService.php:177
/home/greg/dev/SonataBlockBundle/src/Block/Service/MenuBlockService.php:110
/home/greg/dev/SonataBlockBundle/tests/Block/Service/MenuBlockServiceTest.php:83

```

File: src/Block/Service/MenuBlockService.php [hunk 2]
```diff
@@ -161,13 +162,19 @@ protected function getFormSettingsKeys()
     {
         $choiceOptions = [
             'required' => false,
+            'choice_translation_domain' => 'SonataBlockBundle',
         ];
 
         $choices = $this->menuRegistry->getAliasNames();
```
Comment (jordisala1991) [inline src/Block/Service/MenuBlockService.php]: Remove this line

File: src/Block/Service/MenuBlockService.php [hunk 3]
```diff
@@ -161,13 +162,19 @@ protected function getFormSettingsKeys()
...
 
         $choices = array_flip($choices);
```
Comment (jordisala1991) [inline src/Block/Service/MenuBlockService.php]: And this line

File: src/Block/Service/MenuBlockService.php [hunk 4]
```diff
@@ -161,13 +162,19 @@ protected function getFormSettingsKeys()
...
 
-        $choiceOptions['choices'] = $choices;
+        // choice_as_value options is not needed in SF 3.0+
+        if (method_exists(FormTypeInterface::class, 'setDefaultOptions')) {
+            $choiceOptions['choices_as_values'] = true;
+        }
+
+        $choiceOptions['choices'] = array_flip($this->menuRegistry->getAliasNames());
```
Comment (jordisala1991) [inline src/Block/Service/MenuBlockService.php]: Here is where this gets done

**Pull request merged**